core/vm/tosca_integration_test.go

```diff
@@ -0,0 +1,39 @@
+package vm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInterpreterFactory(t *testing.T) {
+	evm := &EVM{}
+	cfg := Config{}
+
+	interpreter := NewInterpreter("", evm, cfg)
+	if interpreter == nil {
+		t.Error("Expected interpreter to be created, but got nil")
+	}
+	interpreter = NewInterpreter("geth", evm, cfg)
+	if interpreter == nil {
+		t.Error("Expected interpreter to be created, but got nil")
+	}
+	asserted := assert.Panics(t, func() { NewInterpreter("invalid", evm, cfg) })
+	if !asserted {
+		t.Error("Expected panic with invalid interpreter name")
+	}
+}
+
+func TestInterpreterFactory_RegisterInterpreter(t *testing.T) {
+	evm := &EVM{}
+	cfg := Config{}
+
+	RegisterInterpreterFactory("newInterpreter", func(evm *EVM, cfg Config) Interpreter {
+		return NewEVMInterpreter(evm)
+	})
+
+	interpreter := NewInterpreter("newInterpreter", evm, cfg)
+	if interpreter == nil {
+		t.Error("Expected interpreter to be created, but got nil")
+	}
```

Comment:
Can you add a local flag that is set from false to true in the factory and which is after the call checked to make sure the correct factory is called?